prow/repoowners/endpoint_test.go

```diff
@@ -0,0 +1,218 @@
+			expectedCode: http.StatusOK,
+		},
+		{
+			name: "Missing content-type information",
+			body: &ownersRequest{
+				Org:        "org",
+				Repo:       "repo",
+				BaseCommit: "Master",
+				Paths: []string{
+					"src/dir",
+					"src",
+					"foo",
+					"src/dir/conformance",
+					"docs",
+					"docs/file.md",
+				},
+			},
+			contentType:  "",
+			expectedBody: `{"error_message":"This JSON-API required a JSON content-type"}`,
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name: "Empty paths",
+			body: &ownersRequest{
+				Org:        "org",
+				Repo:       "repo",
+				BaseCommit: "Master",
+				Paths:      []string{},
+			},
+			contentType:  "application/json",
+			expectedBody: `{"error_message":"Path(s) list is empty or non-existing"}`,
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name: "Non-valid org",
+			body: &ownersRequest{
+				Org:        "9AA48AEB-30EB-4187-89E7-5FAEACC3EC91",
+				Repo:       "repo",
+				BaseCommit: "Master",
+				Paths: []string{
+					"src/dir",
+					"src",
+					"foo",
+					"src/dir/conformance",
+					"docs",
+					"docs/file.md",
+				},
+			},
+			contentType:  "application/json",
+			expectedBody: `{"error_message":"Please provide corret git information for Org/Repo and Base commit"}`,
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name: "Non-valid repo",
+			body: &ownersRequest{
+				Org:        "org",
+				Repo:       "A1A1693C-082C-41B1-9943-3A89B08A57C0",
+				BaseCommit: "Master",
+				Paths: []string{
+					"src/dir",
+					"src",
+					"foo",
+					"src/dir/conformance",
+					"docs",
+					"docs/file.md",
+				},
+			},
+			contentType:  "application/json",
+			expectedBody: `{"error_message":"Please provide corret git information for Org/Repo and Base commit"}`,
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name: "Non-valid basecommit",
+			body: &ownersRequest{
+				Org:        "org",
+				Repo:       "repo",
+				BaseCommit: "13202CD5-3068-46D4-8C1D-46F544ED39EF",
+				Paths: []string{
+					"src/dir",
+					"src",
+					"foo",
+					"src/dir/conformance",
+					"docs",
+					"docs/file.md",
+				},
+			},
+			contentType:  "application/json",
+			expectedBody: `{"error_message":"Please provide corret git information for Org/Repo and Base commit"}`,
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name:         "Empty body",
+			body:         nil,
+			contentType:  "application/json",
+			expectedBody: `{"error_message":"Path(s) list is empty or non-existing"}`,
+			expectedCode: http.StatusBadRequest,
+		},
+	}
+
+	client, cleanup, err := getTestClient(testFiles, true, false, true, nil, nil, nil)
+	if err != nil || client == nil || cleanup == nil {
+		t.Fatalf("Error creating test client: %v.", err)
```

Comment:
This message is not enough. If `err` is `nil` but so is one of the others than I can't get that from this message.